src/MediatR/Pipeline/Streams/StreamRequestExceptionActionProcessorBehavior.cs

```diff
@@ -0,0 +1,64 @@
+namespace MediatR.Pipeline.Streams
+{
+    using MediatR.Internal;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Runtime.CompilerServices;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Behavior for executing all <see cref="IStreamRequestExceptionHandler{TRequest,TResponse,TException}"/>
+    ///     or <see cref="StreamRequestExceptionHandler{TRequest,TResponse}"/> instances
+    ///     after an exception is thrown by the following pipeline steps
+    /// </summary>
+    /// <typeparam name="TRequest">Request type</typeparam>
+    /// <typeparam name="TResponse">Response type</typeparam>
+    public class StreamRequestExceptionActionProcessorBehavior<TRequest, TResponse> : IStreamPipelineBehavior<TRequest, TResponse>
+        where TRequest : notnull
+    {
+        private readonly ServiceFactory _serviceFactory;
+
+        public StreamRequestExceptionActionProcessorBehavior(ServiceFactory serviceFactory) => _serviceFactory = serviceFactory;
+
+        public async IAsyncEnumerable<TResponse> Handle(TRequest request, [EnumeratorCancellation] CancellationToken cancellationToken, StreamHandlerDelegate<TResponse> next)
+        {
+            var asyncEnum = next().WithCancellation(cancellationToken).ConfigureAwait(false).GetAsyncEnumerator();

```

Comment:
If you do the asyncenum dance youreself via movenextasync, aren't you supposed to dispose it yourself too?
The same applies to the behaviour below too.

Reply:
Thanks for the review comment. I will dive into it.